tests/integration/devfile/cmd_devfile_watch_test.go

```diff
@@ -31,105 +31,61 @@ var _ = Describe("odo devfile watch command tests", func() {
 		helper.CommonAfterEach(commonVar)
 	})
 
-	Context("when running help for watch command", func() {
+	When("running help for watch command", func() {
```

Comment:
The important rule is that `When` should describe what is executed into `BeforeEach`.

Here, it is not the case. To stick to the rule, you could either move `odo watch` to `BeforeEach`, or have only one `It` with all the code (by just removing the `When`):

```
It("should display the help", func() {
			appHelp := helper.Cmd("odo", "watch", "-h").ShouldPass().Out()
			helper.MatchAllInOutput(appHelp, []string{"Watch for changes", "git components"})
		})
```

Reply:
Moved odo watch to beforeEach as suggested

Reply:
reverted to just have an it . helper gets undefined if I define appHelp outside the it